code/modules/projectiles/guns/ballistic/revolver.dm

```diff
@@ -304,6 +304,38 @@
 	else
 		to_chat(user, "<span class='warning'>[src] is empty!</span>")
 
+
+// PIPE PISTOL //
+/obj/item/weapon/gun/ballistic/revolver/pipe
+	name = "pipe pistol"
+	desc = "A makeshift 9mm pistol made from salvaged parts." //An engraving on the side reads: Tacolizard Forever
+	icon_state = "pipepistol"
+	w_class = WEIGHT_CLASS_NORMAL
+	mag_type = /obj/item/ammo_box/magazine/internal/shot/makeshift //mag is internal for balance reasons
+	can_suppress = 0
+	sparks = 1
+	dont_update = 1 //hopefully prevents icon from changing to default after a variant is selected
+	slot_flags = SLOT_BACK
+
+
+/obj/item/weapon/gun/ballistic/revolver/pipe/attackby(obj/item/A, mob/user, params)
+	..()
+	if(istype(A, /obj/item/ammo_box) || istype(A, /obj/item/ammo_casing))
```

Comment:
If you're going to add an ammo box for this, you might as well add a crafting recipe for a makeshift 9mm ammo box.